app/models/katello/upstream_consumer.rb

```diff
@@ -0,0 +1,61 @@
+require 'katello/resources/candlepin'
+
+module Katello
+  class UpstreamConsumer < OpenStruct
+    CP_CONSUMER = Resources::Candlepin::UpstreamConsumer
+
+    class << self
+      def bind_entitlements(organization, params)
```

Comment:
Why not just pass pools in directly, instead of through 'params' ?

Reply:
The goal for all this is to have a clear separation of layers, and it seems like there is a lot blurring together between the controller, the model layer, and the api layer.

Reply:
I believe I did it this way because the Candlepin API for bind entitlements is flexible - we can pass in pools, products, times, etc. to bind to. I thought I'd give that flexibility to any consumers of the UpstreamConsumer model. If I change it, I might also change the method name to `bind_entitlements_from_pools` to be specific.

I think it's also confusing because the model layer represents just another API layer actually, so there are params that need to be formatted/deformatted, JSON to parse, etc.